atlp-rwanda/e-commerce-mavericcks-bn
# Configure winston logging library

## Purpose

Winston logging is crucial in projects because it provides a structured and centralized approach to capturing and managing application logs. It helps in debugging, troubleshooting, and monitoring the application's behavior, ultimately improving its reliability and performance.

## Changes Made

- Removed the **logs** folder from the **gitignore** to include it in version control.
- Migrated **@typescript-eslint/parser** and **@typescript-eslint/plugin** to devdependencies for TypeScript linting.
- Removed the **@types/dotenv** package as it's not needed.
- Configured Winston logging in **./src/logs/config** for structured logging.
- Implemented the usage of Winston in the **server.ts** file, adding middleware before and after routes for comprehensive logging (**refer to comments in server.ts for details**).
- Added the **husky** folder to the gitignore to prevent duplication during dependency installation and configured a pre-commit hook to **run npm** test before committing.

## Testing Instructions

To test the changes:

1. Ensure all dependencies are installed (**npm instal**l).
2. Run the tests using **npm test**.
3. Manually verify that logs are being generated in the **logs** folder after running the application.

## Related Issues

None

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:

based on this pull request required information please provide yours based on this and fill unfilled
